src/ApiService/ApiService/onefuzzlib/JobOperations.cs

```diff
@@ -46,17 +44,17 @@ public class JobOperations : StatefulOrm<Job, JobState>, IJobOperations {
         throw new NotImplementedException();
     }
 
-    public async System.Threading.Tasks.Task Stopping(Job job) {
+    public async System.Threading.Tasks.Task Stopping(Job job, ITaskOperations taskOperations) {
```

Comment:
For later, should we just have a "mega-factory" being passed in the constructor, and then each object can create what it needs ? That way it will be harder to have circular dependency ?

Reply:
yep. I was toying with the idea of passing `IServiceCollection` directly. But i think we should have something a bit more specific to Onefuzz

Reply:
Agreed, let's have OneFuzz Factory of some sort, add AddScoped factory, and all creation happens inside of the factory....